portfolio/src/main/java/com/google/sps/data/AUX.java

```diff
@@ -35,51 +48,172 @@ private static void GetDevKey(){
+            String videoCaptions = getVideoCaptions(videoId);
+            Video result = new Video(videoId, videoTitle, videoDescription, thumbnailUrl, videoCaptions, /*isPublic*/ true);
+            return result;
+        }
+        catch (IOException e) {
+            throw new VideoException("Unable to retrieve video");
+        } 
+        catch (IndexOutOfBoundsException e) {
+            throw new VideoException("No such video found.");
+        } 
+        catch (GeneralSecurityException e) {
+            throw new VideoException("A security exception occurred");
+        }
+        catch(Exception e){
+            //Catch any other error that has been potentially overlooked
+            StringWriter sw = new StringWriter();
+            PrintWriter pw = new PrintWriter(sw);
+            if(DEBUG){
+                e.printStackTrace(pw);
+            }
+            throw new VideoException(sw.toString());
+        }
+        
+        
     }
 
-    public static ArrayList<String> SearchQueryToListOfVideoID(String query) throws IOException, GeneralSecurityException{
-        GetDevKey();
+    /*
+    *Returns a list of video ID's based on a search query
+    */
+    public static ArrayList<String> searchQueryToListOfVideoID(String query) throws IOException, GeneralSecurityException{
+        getDevKey();
         YouTube youtubeService = getService();
         // Define and execute the API request
         YouTube.Search.List searchRequest = youtubeService.search()
             .list("snippet")
             .setType("video");
-        SearchListResponse searchResponse = searchRequest.setQ(query)
-            .setKey(DEVELOPER_KEY)
-            .execute();
+
+        SearchListResponse searchResponse = new SearchListResponse();
+        try{
+            searchResponse = searchRequest.setQ(query)
+                .setKey(DEVELOPER_KEY)
+                .execute();
+        }
+        catch(Exception e){
+            if(DEBUG){
+                //Print errors to console
+                StringWriter sw = new StringWriter();
+                PrintWriter pw = new PrintWriter(sw);
+                e.printStackTrace(pw);
+                System.out.println(sw.toString());
+            }
+        }
+        
         ArrayList<String> videoIdList = new ArrayList<String>();
         for(int i = 0; i < searchResponse.getItems().size(); i++){
             videoIdList.add(searchResponse.getItems().get(i).getId().getVideoId());
         }
         return videoIdList;
     }
 
-    /**
-     * Build and return an authorized API client service.
-     *
-     * @return an authorized API client service
-     * @throws GeneralSecurityException, IOException
-     */
-    private static YouTube getService() throws GeneralSecurityException, IOException {
+    /*
+    *Retrieve video id from youtube url
+    *@return a video id as string
+    */
+    public static String youtubeUrlToId(String youtubeUrl)throws IllegalArgumentException{
```

Comment:
Strategy: An approach where we use regular expressions would be more robust and cleaner.
https://regexone.com/ for some tutorials.